src/transaction.js

```diff
@@ -238,7 +222,7 @@ function makeTxClient(trx, client, connection) {
     })
   }
   trxClient.acquireConnection = function() {
-    return trx._queue.then(function() {
+    return Promise.settle([trx._previousSibling]).then(function () {
```

Comment:
This behaviour is unchanged. And from how I read the original code, the idea is that if we have a set of transactions like this:

```
TP (parent)
    - TS1 (first sibling)
    - TS2 (second sibling)
```

TS2 should be allowed to proceed (i.e. acquire its connection so it can perform actual database operations) only after TS1 has completed its work. Whether TS1 got committed or rolled back.

If we want TS2 to be run only if TS1 gets committed, then we should created TS2 in a `.then()` block chained to the TS1 resolution promise, and not just create it as a completely unrelated sibling transaction.